cosmic-settings/src/pages/networking/wifi.rs

```diff
@@ -0,0 +1,668 @@
+                        network_manager::watch_connections_changed(conn, tx)
+                    );
+                },
+            ));
+        }
+    }
+
+    /// Closes the view more popup and applies any withheld updates.
+    fn close_popup_and_apply_updates(&mut self) {
+        self.view_more_popup = None;
+        if let Some(ref mut nm_state) = self.nm_state {
+            if let Some(state) = self.withheld_state.take() {
+                nm_state.state = state;
+            }
+
+            if let Some(devices) = self.withheld_devices.take() {
+                nm_state.devices = devices;
+            }
+        }
+    }
+
+    /// Withholds updates if the view more popup is displayed.
+    fn update_devices(&mut self, devices: Vec<network_manager::devices::DeviceInfo>) {
+        if let Some(ref mut nm_state) = self.nm_state {
+            if self.view_more_popup.is_some() {
+                self.withheld_devices = Some(devices);
+            } else {
+                nm_state.devices = devices;
+            }
+        }
+    }
+
+    /// Withholds updates if the view more popup is displayed.
+    fn update_state(&mut self, state: NetworkManagerState) {
+        if let Some(ref mut nm_state) = self.nm_state {
+            if self.view_more_popup.is_some() {
+                self.withheld_state = Some(state);
+            } else {
+                nm_state.state = state;
+            }
+        }
+    }
+}
+
+fn devices_view() -> Section<crate::pages::Message> {
+    crate::slab!(descriptions {
+        airplane_mode_txt = fl!("airplane-on");
+        connect_txt = fl!("connect");
+        connected_txt = fl!("connected");
+        connecting_txt = fl!("connecting");
+        disconnect_txt = fl!("disconnect");
+        forget_txt = fl!("wifi", "forget");
+        known_networks_txt = fl!("known-networks");
+        no_networks_txt = fl!("no-networks");
+        settings_txt = fl!("settings");
+        visible_networks_txt = fl!("visible-networks");
+        wifi_txt = fl!("wifi");
+    });
+
+    Section::default()
+        .descriptions(descriptions)
+        .view::<Page>(move |_binder, page, section| {
+            let Some(NmState { ref state, .. }) = page.nm_state else {
+                return cosmic::widget::column().into();
+            };
+
+            let theme = cosmic::theme::active();
+            let spacing = &theme.cosmic().spacing;
+
+            let wifi_enable =
+                widget::settings::item::builder(&section.descriptions[wifi_txt]).control(
+                    widget::toggler(None, state.wifi_enabled, Message::WiFiEnable),
+                );
+
+            let mut view = widget::column::with_capacity(4)
+                .push(widget::list_column().add(wifi_enable))
+                .push_maybe(state.airplane_mode.then(|| {
+                    widget::container(widget::text::body(&section.descriptions[airplane_mode_txt]))
+                        .align_x(alignment::Horizontal::Center)
```

Comment:
```suggestion
widget::container(
    widget::row::with_capacity(2)
        .push(icon::from_name("airplane-mode-symbolic"))
        .push(widget::text::body(&section.descriptions[airplane_mode_txt]))
        .spacing(8)
        .align_items(alignment::Alignment::Center),
)
.width(Length::Fill)
.align_x(alignment::Horizontal::Center)
```

`.width(Length::Fill)` is also needed in the `no_networks_found` element below.